TRUNK-4988: Added MessageService for Spring method injection

snowvirus opened the pull request:
<!--- Add a pull request title above in this format -->
<!--- real example: 'TRUNK-5111: Replace use of deprecated isVoided' -->
<!--- 'TRUNK-JiraIssueNumber: JiraIssueTitle' -->
## Description of what I changed
<!--- Describe your changes in detail -->
<!--- It can simply be your commit message, which you must have -->

## Issue I worked on
<!--- This project only accepts pull requests related to open issues -->
<!--- Want a new feature or change? Discuss it in an issue first! -->
<!--- Found a bug? Point us to the issue/or create one so we can reproduce it! -->
<!--- Just add the issue number at the end: -->
see https://issues.openmrs.org/browse/TRUNK-

## Checklist: I completed these to help reviewers :)
<!--- Put an `x` in the box if you did the task -->
<!--- If you forgot a task please follow the instructions below -->
- [x] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [x] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [x] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [x] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [x] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

snowvirus commented:
am requesting for a review. and some guidlines on how to do it in a good way. if at all everything i did was nothing 
Thank you
@dkayiwa @ibacher 
issue: https://openmrs.atlassian.net/browse/TRUNK-4988
and you can also help me on how to do the PR in a correct way. And what is needed in it

dkayiwa commented:
Did you get a chance to look at this? http://openmrs.atlassian.net/wiki/display/docs/Pull+Request+Tips

snowvirus commented:
> Did you get a chance to look at this? http://openmrs.atlassian.net/wiki/display/docs/Pull+Request+Tips

i think i did that.  http://openmrs.atlassian.net/wiki/display/docs/Pull+Request+Tips 
now the a review

snowvirus commented:
Anyone to review my work if possible? @ibacher @dkayiwa

ibacher commented:
@snowvirus Could you please feel out the information in the PR template? It helps actually review your request. It would also be helpful if you could explain how this PR actually matches what the ticket is asking for.

snowvirus commented:
Description:
I refactored Context.java to use Spring’s method injection for setting dependencies on the MessageService.
This resolves the TODOs on lines 560 and 570 of Context.java.
I added/modifed MessageService.java accordingly.
Issue Link:issue: https://openmrs.atlassian.net/browse/TRUNK-4988

How This PR Solves the Issue:
The issue requested Spring’s method injection to be used instead of manually managing dependencies.
I implemented the necessary setter methods to allow dependency injection for MessageService.
I verified the functionality by running the existing tests and adding new ones.
Testing:
I ran mvn clean package, mvn clean install and mvn test, and all tests passed/build successful. but a few tests were skipped, but others passed.
@ibacher  am request for a review
@dkayiwa

snowvirus commented:
> Description: I refactored Context.java to use Spring’s method injection for setting dependencies on the MessageService. This resolves the TODOs on lines 560 and 570 of Context.java. I added/modifed MessageService.java accordingly. Issue Link:issue: https://openmrs.atlassian.net/browse/TRUNK-4988
> 
> How This PR Solves the Issue: The issue requested Spring’s method injection to be used instead of manually managing dependencies. I implemented the necessary setter methods to allow dependency injection for MessageService. I verified the functionality by running the existing tests and adding new ones. Testing: I ran mvn clean package, mvn clean install and mvn test, and all tests passed/build successful. but a few tests were skipped, but others passed. @ibacher am request for a review @dkayiwa

review request. @ibacher @dkayiwa @michaelpigg

ibacher commented:
Please put the update in the PR description. Also, please re-read the JIRA issue carefully. This creates a service, but by itself, this isn't very useful. The JIRA issue is about an already existing service. Please re-read the issue, think about what it's asking you to do. If you're not clear on that, I'd suggest picking another issue to start from. If you're having questions about how to go about implementing the ticket, the right place to discuss it is either on Talk or Slack not here.

I'm going to close this PR because it needs to be completely re-written to match the ticket, but feel free to reopen once it's clearer what the ticket is actually asking for.
